src/Sentry/Internal/BatchProcessor.cs

```diff
@@ -0,0 +1,99 @@
+using System.Timers;
+using Sentry.Protocol;
+using Sentry.Protocol.Envelopes;
+
+#if NET9_0_OR_GREATER
+using Lock = System.Threading.Lock;
+#else
+using Lock = object;
+#endif
+
+namespace Sentry.Internal;
+
+/// <summary>
+/// The Sentry Batch Processor.
+/// This implementation is not complete yet.
+/// Also, the specification is still work in progress.
+/// </summary>
+/// <remarks>
+/// Sentry Specification: <see href="https://develop.sentry.dev/sdk/telemetry/spans/batch-processor/"/>.
+/// OpenTelemetry spec: <see href="https://github.com/open-telemetry/opentelemetry-collector/blob/main/processor/batchprocessor/README.md"/>.
+/// </remarks>
+internal sealed class BatchProcessor : IDisposable
+{
+    private readonly IHub _hub;
+    private readonly BatchProcessorTimer _timer;
+    private readonly BatchBuffer<SentryLog> _logs;
+    private readonly Lock _lock;
+
+    private DateTime _lastFlush = DateTime.MinValue;
+
+    public BatchProcessor(IHub hub, int batchCount, TimeSpan batchInterval)
+        : this(hub, batchCount, new TimersBatchProcessorTimer(batchInterval))
+    {
+    }
+
+    public BatchProcessor(IHub hub, int batchCount, BatchProcessorTimer timer)
+    {
+        _hub = hub;
+
+        _timer = timer;
+        _timer.Elapsed += IntervalElapsed;
+
+        _logs = new BatchBuffer<SentryLog>(batchCount);
+        _lock = new Lock();
+    }
+
+    internal void Enqueue(SentryLog log)
+    {
+        lock (_lock)
+        {
+            EnqueueCore(log);
+        }
+    }
+
+    private void EnqueueCore(SentryLog log)
+    {
+        var empty = _logs.IsEmpty;
```

Comment:
Suggest renaming this variable to `isFirstLog`... we're determining that by checking whether `_logs` was empty before adding the log but ultimately that's what we care about.

Later on, this then becomes a bit more readable/obvious:
```csharp
        if (isFirstLog && !_logs.IsFull)
        {
            _timer.Enabled = true;
        }
```